### ISSUE TITLE
Documentation about upgrading the docker version

### ISSUE THREAD

**Issue by nathanleclaire** [docker/docker#9461]
I might take a crack at this if there's interest, but I'd like to at least note down that I'm keen on having a page in documentation dedicated to discussing the upgrade path for users running versions of Docker that lag behind the latest.  In particular it would be useful to formalize what to do in the case of when the user has installed with `curl get.docker.com | sh` and they want to upgrade.  Especially when urgent security fixes come out, it'd be nice to have a canonical way to do this (and possibly a script or something that does much of the work for you).

Right now what I do for hobby use is:

1. download the latest binary i.e. https://docs.docker.com/installation/binaries/#get-the-docker-binary
2. replace the existing binary 
3. `sudo service docker restart`
4. `docker run` the containers that were previously running again.  I can't remember if the processes that were run when the daemon restart happens get ghosted (which might cause odd issues), but they didn't show up in `docker ps` after a daemon reset with a new version last time that I checked.  I'm not sure if this is expected behavior or not.

My way of doing things might be completely broken but without an official word from us I'm not entirely sure if it is or isn't.

Also, if you're running a version from a package manager (say, 1.2) that is vulnerable and the official package archives lag behind the latest docker release, what do you do?

**Comment by andrevtg** [comment]
+1

I can tell for sure that Reh Hat/CentOS yum upgrade has been disastrous for my servers - I've lost all my running containers and the daemon could run again only after wiping out "/var/lib/docker".

It was not a big deal for me because deployment was automated with Ansible, but I could live without such hiccups...

**Comment by thaJeztah** [comment]
@andrevtg I'm not sure we can cover every situation. Testing upgrades before rolling them out to production is important, but may be out of scope for our documentation.


> Also, if you're running a version from a package manager (say, 1.2) that is vulnerable and the official package archives lag behind the latest docker release, what do you do?

I'm not sure this is still a concern. We provide our own packages for all major distros, and the security team makes sure that maintainers of official packages in those distros are informed early on vulnerabilities.


We currently have upgrading instructions in the Ubuntu documentation; http://docs.docker.com/engine/installation/ubuntulinux/#upgrade-docker

But we may add similar instructions on the RHEL/CentOS/Fedora pages.

I also just opened an issue on documenting how to install a specific version and/or "pin" to a version; https://github.com/docker/docker/issues/17935

**Comment by nathanleclaire** [comment]
This issue p. old, I think we can close since it's not been taken action on.  For Debian packages at least `sudo apt-get install -y docker-engine` gets the job done in a pretty predictable fashion
